pyccel/parser/semantic.py

```diff
@@ -1114,6 +1135,26 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
 
+                if isinstance(lhs, DottedName):
+                    prefix = self._visit(DottedName(*lhs.name[:-1]))
+                    class_def = prefix.cls_base
+                    if isinstance(class_def, ClassDef) and lhs.name[0] == 'self':
+                        var      = self.get_variable('self')
+
+                        name = self.scope.get_expected_name(str(lhs.name[-1]))
+                        member = self._create_variable(name, dtype, rhs, d_lhs)
+
+                        lhs    = member.clone(member.name, new_class = DottedVariable, lhs = var)
+                        if not str(lhs.name[-1]) == name:
+                            self.scope.remove_variable(lhs, lhs.name[-1])
+                            self.scope.insert_variable(lhs, name)
+
+                        # update the attributes of the class and push it to the scope
+                        class_def.add_new_attribute(member)
+                        self.scope.parent_scope.update_class(class_def)
```

Comment:
```suggestion
                    if isinstance(class_def, ClassDef) and lhs.name[0] == 'self':
                        var      = self.get_variable('self')
                        
                        # Get the scope for the class
                        class_scope = class_def.scope

                        # Collect the name that should be used in the generated code
                        attribute_name = lhs.name[-1]
                        new_name = class_scope.get_expected_name(attribute_name)
                        # Create the attribute
                        member = self._create_variable(name, dtype, rhs, d_lhs)
                        
                        # Insert the attribute to the class scope
                        # Passing the original name ensures that the attribute can be found under this name 
                        class_def.scope.insert_variable(member, attribute_name)

                        # Create the local DottedVariable
                        lhs    = member.clone(member.name, new_class = DottedVariable, lhs = var)

                        # update the attributes of the class and push it to the scope
                        class_def.add_new_attribute(member)
```

Reply:
Fixed in https://github.com/pyccel/pyccel/pull/1484/commits/975113feafc9e9351e2780c515b649aaf9c02030.
But I added a conditions cause the `insert_variable` returns an Error `Variable already exists in scope` and I m investigating the source of the Error.